File: easy/ts/1732.ts
// https://leetcode.com/problems/find-the-highest-altitude/?envType=study-plan-v2&envId=leetcode-75
// Time: O(n)
// Space: O(1)
function largestAltitude(gain: number[]): number {
  let max = 0;
  let currentVal = 0;

  for (let i = 0; i < gain.length; i++) {
    currentVal += gain[i];
    if (currentVal > max) {
      max = currentVal;
    }
  }
  return max;
}

File: easy/python/1732.py
# https://leetcode.com/problems/find-the-highest-altitude/description/?envType=study-plan-v2&envId=leetcode-75
# Time: O(n)
# Space: O(1)
class Solution:
    def largestAltitude(self, gain: List[int]) -> int:
        max_val = 0
        current_val = 0

        for ele in gain:
            current_val += ele
            if current_val > max_val:
                max_val = current_val
        return max_val
